lib/storage/file.js

```diff
@@ -366,8 +384,20 @@ File.prototype.copy = function(destination, callback) {
   var newFile;
 
   if (is.string(destination)) {
-    destBucket = this.bucket;
-    destName = destination;
+    var parsedDestination = GS_URL_REGEXP.exec(destination);
+    if (parsedDestination !== null && parsedDestination.length === 3) {
+      // We can't access the Bucket constructor without an infinite `require`
+      // loop, so use Object.create with this file's bucket.
+      destBucket = Object.create(this.bucket, {
```

Comment:
You should be able to get the parent Storage instance and create a new bucket like this:

```js
var storageInstance = this.bucket.parent;
destBucket = storageInstance.bucket(parsedDestination[1]);
destName = parsedDestination[2];
```